lightning-invoice/src/utils.rs

```diff
@@ -365,7 +336,7 @@ where
 		.duration_since_epoch(duration_since_epoch)
 		.payee_pub_key(our_node_pubkey)
 		.payment_hash(Hash::from_slice(&payment_hash.0).unwrap())
-		.payment_secret(payment_secret)
+		.payment_secret(payment_secret.unwrap())
```

Comment:
noob question again... I am trying to do what you said as 
```
		.payment_secret(payment_secret)
		.map_err(|()| SignOrCreationError::CreationError(CreationError::InvalidAmount))?
```
but its yelling at me with the same thing where it doesn't like how payment_secret is throwing result<PaymentSecret>. What am I not understanding? Once I do this will try to squash and then push again.

Reply:
ok great, I think I have done that. Pushed again, hopefully I did the rebase correctly as well? Let me know if there are any issues or further changes needed.